client/my-sites/plans-features-main/index.tsx

```diff
@@ -726,6 +732,15 @@ const PlansFeaturesMain = ( {
 	);
 
 	const isPlansGridReady = ! isLoadingGridPlans && ! resolvedSubdomainName.isLoading;
+	const [ comparisonGridPlanTypeSelectorRef, isComparisonGridPlanTypeSelectorInView ] = useInView( {
+		rootMargin: '0px 0px -300px 0px',
+	} );
+
+	const stickyPlanTypeSelectorHeight = 48;
+	const comparisonGridStickyRowOffset =
+		showPlanSelectorDropdown && isMobile()
```

Comment:
**Note:**

@chriskmnds I know that you mentioned wanting to keep responsive desktop / tablet / mobile concerns to css media query calculations ( and avoid the use of `isMobile` ) https://github.com/Automattic/wp-calypso/pull/85158#discussion_r1424955463

I did a decent amount of tinkering, but I couldn't figure out a clear way to proceed with media query mixins for this specific case without adding a non-trivial amount of complexity. The sticky offset we'd like to adjust depends on `masterbarHeight`, which is calculated at runtime by our javascript, and inaccessible for evaluation in CSS.

I might be missing something obvious though since I've been staring at this code for quite a while now 🤔 Totally open to suggestions.

Reply:
Thanks @jeyip . I had a brief look at the code in general and it looks pretty good. I have some comments for reflection, which i'll post later.

For this case, if we do end up with unreasonable complexity, then fine to use other means. Also, if the logic stays outside `plans-grid`, then we don't really care about how we define "mobile". The plans-grid package will likely have its own "small", "medium", "large" dimensions that will be determined differently (through own listener), but anything that consumes these media queries will just use those props instead (hence why best to not introduce another way). I need to take a closer look - will reply later in detail.